### PR TITLE
 GearKnight Invasion Direlands & Inner Sea

### PR DESCRIPTION
**Pull request by DridsGhost**
This builds off of Tectonic Rifts work.

Initial update to killtasks and other emote rewrites.

Added additional quest entries.

Content ready to be tested

http://acpedia.org/wiki/Direlands_Gear_Knight_Resistance_Camp

Route: Give Merkitz ibn'Akadh in Xarabydun one MMD note to summon a portal. You could also run from Walled Portals or Fort Tethana

Apostate Finale
http://acpedia.org/wiki/Apostate_Finale

Aetherium Ore Collection
http://acpedia.org/wiki/Aetherium_Ore_Collection

Atamarr Tasks
http://acpedia.org/wiki/Atamarr_Tasks

Aetherium Raid (High)
http://acpedia.org/wiki/Aetherium_Raid_(High)

Gear Knight Kill Task
http://acpedia.org/wiki/Gear_Knight_Kill_Task

Iron Blade Commander
http://acpedia.org/wiki/Iron_Blade_Commander

-Mana Siphon Destruction (To be implemented with Societies Patch)

-Gear Knight Construction (Not writing this huge quest/emote logic. Someobdy else can volunteer.)

INNER SEA

Jin Ato - All Complete
https://asheron.fandom.com/wiki/Inner_Sea_Aetherium_Collection

Lieutenant Micham - All Complete
http://acpedia.org/wiki/Iron_Blade_Trooper_Kill_Task

 Lieutenant Grenlin - All Complete
http://acpedia.org/wiki/Bronze_Gauntlet_Trooper_Kill_Task

Lieutenant Zin - All Complete
http://acpedia.org/wiki/Copper_Cog_Trooper_Kill_Task
 
Lieutenant Faen - All Complete
https://asheron.fandom.com/wiki/Silver_Scope_Trooper_Kill_Task

Sir Stavitor - All Complete
https://asheron.fandom.com/wiki/Gold_Gear_Trooper_Kill_Task

Sir Oakley - All Complete (Funky emote)
https://asheron.fandom.com/wiki/Aetherium_Raid_(Low)

Dame Alistra - All Complete
https://asheron.fandom.com/wiki/Gearknight_Lesser_Lords

### CODE REVIEW

File: Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Book/44984 Large Tome.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+INSERT INTO `weenie` (`class_Id`, `class_Name`, `type`, `last_Modified`)
+VALUES (44984, 'ace44984-largetome', 8, '2020-02-08 00:01:31') /* Book */;
+
+INSERT INTO `weenie_properties_int` (`object_Id`, `type`, `value`)
+VALUES (44984,   1,       8192) /* ItemType - Writable */
+     , (44984,   5,        100) /* EncumbranceVal */
+     , (44984,  16,          8) /* ItemUseable - Contained */
+     , (44984,  19,          0) /* Value */
+     , (44984,  33,          1) /* Bonded - Bonded */
+     , (44984,  53,        101) /* PlacementPosition - Resting */
```
Comment (ziang4891) [inline Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Book/44984 Large Tome.sql:12]: **PlacementPosition - Resting** is unnecessary in the weenie file, as it is the default for PropertyInt 53

File: Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Book/44984 Large Tome.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     , (44984,  93,       1044) /* PhysicsState - Ethereal, IgnoreCollisions, Gravity */
+     , (44984, 114,          1) /* Attuned - Attuned */
+     , (44984, 174,          1) /* AppraisalPages */
+     , (44984, 175,          1) /* AppraisalMaxPages */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (44984,   1, True ) /* Stuck */
```
Comment (ziang4891) [inline Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Book/44984 Large Tome.sql:19]: Only Stuck, Inscribable, and IsSellable are needed, as Attackable true is the default, and the remaining properties are set by PropertyInt.PhysicsState

File: Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Creature/GearKnight/41527 Invading Bronze Gauntlet Knight.sql [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+VALUES (41527,   1,         16) /* ItemType - Creature */
+     , (41527,   2,         99) /* CreatureType - GearKnight */
+     , (41527,   6,        255) /* ItemsCapacity */
+     , (41527,   7,        255) /* ContainersCapacity */
+     , (41527,  16,          1) /* ItemUseable - No */
+     , (41527,  25,        185) /* Level */
+     , (41527,  93,       1032) /* PhysicsState - ReportCollisions, Gravity */
+     , (41527, 133,          2) /* ShowableOnRadar - ShowMovement */
+     , (41527, 146,     800000) /* XpOverride */
+     , (41527, 307,          5) /* DamageRating */;
```
Comment (ziang4891) [inline Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Creature/GearKnight/41527 Invading Bronze Gauntlet Knight.sql:16]: DamageRating of 5 shown in the PCAP is actually provided by their weapon.  Having this property here would effectively cause the mob to have a final damage rating of 10.  For mobs that are weaponless, the DamageRating shown in the PCAP would be accurate to added to the mob weenie itself.

File: Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Gem/Gem/41528 Aetherium Power Core.sql [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+     , (41528,   5,        100) /* EncumbranceVal */
+     , (41528,  11,         10) /* MaxStackSize */
+     , (41528,  12,          1) /* StackSize */
+     , (41528,  16,          1) /* ItemUseable - No */
+     , (41528,  19,          0) /* Value */
+     , (41528,  33,          1) /* Bonded - Bonded */
+     , (41528,  93,       1044) /* PhysicsState - Ethereal, IgnoreCollisions, Gravity */
+     , (41528, 114,          1) /* Attuned - Attuned */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
```
Comment (ziang4891) [inline Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Gem/Gem/41528 Aetherium Power Core.sql:17]: Only IsSellable should be needed as the other entries are either provided by PropertyInt.PhysicsState or are default

Comment (ziang4891) [reply Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Gem/Gem/41528 Aetherium Power Core.sql:17]: Same comment or similar commenting goes for all other items in this PR

**Comment by DridsGhost** [comment]
Did anybody test this, content wise? Nuances aside, this ran on drunkenfells test server for awhile, seems to be pretty solid.

**Comment by bobigor377** [comment]
Haven't fully tested it yet, but one thing that will need updated before merge is the mob aggro ranges, similar to the update done for Neftet in #415

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/2170.sql [hunk 5]
```diff
@@ -0,0 +1,66 @@
+DELETE FROM `landblock_instance` WHERE `landblock` = 0x2170;
+
```
Comment (LtRipley36706) [inline Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/2170.sql:2]: the guids for this landblock are all out out of range
[see herel](https://github.com/ACEmulator/ACE-PCAP-Exports/blob/master/Database/3-Core/6%20LandBlockExtendedData/SQL/2170.sql), in addition, technically speaking looks like most stuff was perhaps on a generator?

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/2176.sql [hunk 6]
```diff
@@ -0,0 +1,162 @@
+DELETE FROM `landblock_instance` WHERE `landblock` = 0x2176;
+
```
Comment (LtRipley36706) [inline Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/2176.sql:2]: same guid issues as above
[guids](https://github.com/ACEmulator/ACE-PCAP-Exports/blob/master/Database/3-Core/6%20LandBlockExtendedData/SQL/2176.sql)

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8755.sql [hunk 7]
```diff
@@ -0,0 +1,38 @@
+DELETE FROM `landblock_instance` WHERE `landblock` = 0x8755;
+
+INSERT INTO `landblock_instance` (`guid`, `weenie_Class_Id`, `obj_Cell_Id`, `origin_X`, `origin_Y`, `origin_Z`, `angles_W`, `angles_X`, `angles_Y`, `angles_Z`, `is_Link_Child`, `last_Modified`)
+VALUES (0x787551, 41567, 2270494742, 62.7562, 126.048, 10.005, 0.16201, 0, 0, 0.9867891, False, '2019-11-15 00:00:00');
```
Comment (LtRipley36706) [inline Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8755.sql:4]: [guids](https://github.com/ACEmulator/ACE-PCAP-Exports/blob/master/Database/3-Core/6%20LandBlockExtendedData/SQL/8755.sql)

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8850.sql [hunk 8]
```diff
@@ -0,0 +1,14 @@
+DELETE FROM `landblock_instance` WHERE `landblock` = 0x8850;
+
+INSERT INTO `landblock_instance` (`guid`, `weenie_Class_Id`, `obj_Cell_Id`, `origin_X`, `origin_Y`, `origin_Z`, `angles_W`, `angles_X`, `angles_Y`, `angles_Z`, `is_Link_Child`, `last_Modified`)
+VALUES (0x788501, 7924, 2286944304, 132.107, 180.053, 31.5189, -4.371139e-08, 0, 0, -1, False, '2019-11-15 00:00:00');
```
Comment (LtRipley36706) [inline Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8850.sql:4]: guids

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8851.sql [hunk 9]
```diff
@@ -0,0 +1,67 @@
+DELETE FROM `landblock_instance` WHERE `landblock` = 0x8851;
+
+INSERT INTO `landblock_instance` (`guid`, `weenie_Class_Id`, `obj_Cell_Id`, `origin_X`, `origin_Y`, `origin_Z`, `angles_W`, `angles_X`, `angles_Y`, `angles_Z`, `is_Link_Child`, `last_Modified`)
```
Comment (LtRipley36706) [inline Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8851.sql:3]: guids

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8852.sql [hunk 10]
```diff
@@ -0,0 +1,257 @@
+DELETE FROM `landblock_instance` WHERE `landblock` = 0x8852;
+
+INSERT INTO `landblock_instance` (`guid`, `weenie_Class_Id`, `obj_Cell_Id`, `origin_X`, `origin_Y`, `origin_Z`, `angles_W`, `angles_X`, `angles_Y`, `angles_Z`, `is_Link_Child`, `last_Modified`)
+VALUES (0x00788521, 70699, 0x8852032A, 73.8958, 97.8989, -17.595, 0.999977, 0, 0, -0.006856, False, '2019-11-15 00:00:00'); /* Aetherium Raid Low Gen */
```
Comment (LtRipley36706) [inline Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8852.sql:4]: guids

File: Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Creature/GearKnight/41527 Invading Bronze Gauntlet Knight.sql (hunk 3 above)
Comment (DridsGhost) [reply Database/Patches/2009-10-GearsOfChange/9 WeenieDefaults/Creature/GearKnight/41527 Invading Bronze Gauntlet Knight.sql]: Damage ratings removed.

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/2170.sql (hunk 5 above)
Comment (DridsGhost) [reply Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/2170.sql:2]: fixed

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8755.sql (hunk 7 above)
Comment (DridsGhost) [reply Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8755.sql]: fixed

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8850.sql (hunk 8 above)
Comment (DridsGhost) [reply Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8850.sql]: fixed

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8851.sql (hunk 9 above)
Comment (DridsGhost) [reply Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8851.sql:3]: fixed

File: Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8852.sql (hunk 10 above)
Comment (DridsGhost) [reply Database/Patches/2009-10-GearsOfChange/6 LandBlockExtendedData/8852.sql]: fixed

**Pull request merged**